qcelemental/molparse/chgmult.py

```diff
@@ -490,28 +498,29 @@ def stringify(start, final):
     fm_text = ', '.join((stringify(fs, ff) for fs, ff in zip(fragment_multiplicities, fm_final)))
 
     brief = []
-    brief.append('    {:26} {}'.format('      charge = ' + c_text, 'fragments = ' + fc_text))
-    brief.append('    {:26} {}'.format('multiplicity = ' + m_text, 'fragments = ' + fm_text))
+    if log_brief:
+        brief.append('    {:26} {}'.format('      charge = ' + c_text, 'fragments = ' + fc_text))
+        brief.append('    {:26} {}'.format('multiplicity = ' + m_text, 'fragments = ' + fm_text))
 
     been_defaulted = []
     if c_text.count('(') + fc_text.count('(') > 1:
         been_defaulted.append('charge')
     if '(' in m_text or '(' in fm_text:
         been_defaulted.append('multiplicity')
 
-    if been_defaulted:
+    if been_defaulted and log_brief:
         brief.append(
             '    Note: Default values have been applied for {}. Specify intentions in molecule input block'.format(
                 ' and '.join(been_defaulted)))
 
-    if m_final != _high_spin_sum(fm_final):
+    if (m_final != _high_spin_sum(fm_final)) and log_brief:
         brief.append(
             '    Warning: Total multiplicity is not high-spin sum of fragments; may be clobbered by psi4.core.Molecule.update_geometry().'
         )
 
-    if verbose >= 2:
+    if log_full:
```

Comment:
yikes, was forming those strings and appending them to `brief` or `text` slowing things down?

Reply:
Yea, about 30% of the total time or ~3 seconds in the above example.